build-scripts/component-common.gradle

```diff
@@ -50,3 +50,115 @@ dependencies {
     androidTestImplementation libs.test.espresso.core
     androidTestImplementation libs.test.runner
 }
+
+// Shared logic for projects that depend on libmegazord
+//
+// This ensures that libmegazord will be in the library path so that it can be loaded.  It also adds
+// the transitive JNA dependency.
+ext.dependsOnTheMegazord = {
+    dependencies {
+        api project(":full-megazord")
+
+        // Add a JNA dependency, which is required by UniFFI.
+        //
+        // The tricky thing is getting the correct libjnidispatch library. Use the AAR version for
+        // normal builds, since that ensures the various Android libraries get packaged correctly when
+        // we build our own AAR.  Use the JAR version for tests, since that ensures the tests can load
+        // the native version.
+        implementation(libs.jna) {
+            artifact {
+                type = "aar"
+            }
+        }
+        testImplementation(libs.jna) {
+            artifact {
+                type = "jar"
+            }
+        }
+    }
+
+    // Configurations are a somewhat mysterious Gradle concept.  For our purposes, we can treat them
+    // sets of files produced by one component and consumed by another.
+    configurations {
+        megazordNative {
+            canBeConsumed = false
+        }
+    }
+
+    // Wire up the megazordNative configuration to the output produced from the `full-megazord` project.
+    dependencies {
+        megazordNative(project("path": ":full-megazord", "configuration": "megazordNative"))
+    }
+
+    afterEvaluate {
+        android.libraryVariants.all { variant ->
+            def variantName = variant.name.capitalize();
+            def testTask = tasks["test${variantName}UnitTest"]
+            def processTestResTask = tasks["process${variantName}UnitTestJavaRes"]
+
+            // Copy the native libmegazord to the resource dir so that it can be loaded by JNA.
+            // Note: we have to manually copy the library to the output dir ourselves.  If we simply
+            // added the megazordNative directory to sourceSets.test.resources.srcDirs, then the
+            // android gradle plugin will refuse to copy it. For details see:
+            //
+            // * https://github.com/mozilla/application-services/pull/6476#issuecomment-2537227576
+            // * https://github.com/mozilla/glean/pull/2680#issuecomment-2056627683
+            def copyNativeMegazord = tasks.register("copy${variantName}NativeMegazord", Copy) {
+                from configurations.getByName("megazordNative")
+                into processTestResTask.destinationDir
+                // Make sure to run after the process java res task, otherwise that one may
+                // overwrite our work.
+                dependsOn(processTestResTask)
+            }
```

Comment:
I think this task is the key to getting the AGP upgrade to work.  Something similar should also work for Glean @badboy

Reply:
Yeah, that does feel like it's likely to be an issue later on.

Your idea could work.  Another idea I had was to actually build a `.jar` file with the native libmegazord and depend on that -- that's how we're loading `libjnidispatch` after all.  Maybe we can explore these ideas more in the future.